http3/frame_reader.go

```diff
@@ -0,0 +1,97 @@
+package http3
+
+import (
+	"io"
+
+	"github.com/lucas-clemente/quic-go/quicvarint"
+)
+
+// FrameReader implements a lightweight HTTP/3 frame reader.
+// After setting R, an otherwise zero-value FrameReader is ready to use.
+// If a frame has already been partially read from R, a FrameReader
+// can be initialized by setting Type to the current frame type and N to
+// the number of bytes remaining in the payload.
+type FrameReader struct {
```

Comment:
Why is this exported? What was wrong with the `parseNextFrame` function?

Reply:
`FrameReader` allows the caller to determine which frames to skip, and makes no attempt to parse the underlying frames. This was necessary because different stream types (and perspectives) support specific frame types, and the spec(s) require closing the stream or connection in the event of certain frame types.

Reply:
It’s exported so callers implementing lower-level parts of the HTTP/3 protocol could use the same mechanism. In theory, this could go in a `http3guts` package.